Mathlib/Data/Int/Cast/Basic.lean

```diff
@@ -7,59 +7,141 @@ import Mathlib.Data.Int.Cast.Defs
-`int_cast : ℤ → R` field.
+This file proves additional properties about the *canonical* homomorphism from
+the integers into an additive group with a one (`int.cast`).
 
-Preferentially, the homomorphism is written as a coercion.
+There is also `Data.Int.Cast.Lemmas`,
+which includes lemmas stated in terms of algebraic homomorphisms,
+and results involving the order structure of `ℤ`.
 
-## Main declarations
-
-* `int.cast`: Canonical homomorphism `ℤ → R`.
-* `add_group_with_one`: Type class for `int.cast`.
+By contrast, this file's only import beyond `Data.Int.Cast.Defs` is `Algebra.Group.Basic`.
 -/
 
+
+universe u
+
 namespace Nat
-variable [AddGroupWithOne R]
+
+variable {R : Type u} [AddGroupWithOne R]
 
 @[simp, norm_cast]
 theorem cast_sub {m n} (h : m ≤ n) : ((n - m : ℕ) : R) = n - m :=
   eq_sub_of_add_eq <| by rw [← cast_add, Nat.sub_add_cancel h]
+#align nat.cast_sub Nat.cast_subₓ
+
+@[simp, norm_cast]
+theorem cast_pred : ∀ {n}, 0 < n → ((n - 1 : ℕ) : R) = n - 1
+  | 0, h => by cases h
+  | n + 1, _ => by rw [cast_succ, add_sub_cancel]; rfl
+#align nat.cast_pred Nat.cast_pred
 
 end Nat
 
+open Nat
+
 namespace Int
-variable [AddGroupWithOne R]
+
+variable {R : Type u} [AddGroupWithOne R]
+
+@[simp, norm_cast]
+theorem cast_negSucc (n : ℕ) : (-[n+1] : R) = -(n + 1 : ℕ) :=
+  AddGroupWithOne.intCast_negSucc n
+#align int.cast_neg_succ_of_nat Int.cast_negSuccₓ
+
+@[simp, norm_cast]
+theorem cast_zero : ((0 : ℤ) : R) = 0 :=
+  (AddGroupWithOne.intCast_ofNat 0).trans Nat.cast_zero
+#align int.cast_zero Int.cast_zeroₓ
+
+@[simp high, nolint simpNF] -- this lemma competes with `Int.ofNat_eq_cast` to come later
+theorem cast_ofNat (n : ℕ) : ((n : ℤ) : R) = n :=
+  AddGroupWithOne.intCast_ofNat _
+#align int.cast_coe_nat Int.cast_ofNatₓ
+
+@[simp, norm_cast]
+theorem cast_one : ((1 : ℤ) : R) = 1 := by
+  erw [cast_ofNat, Nat.cast_one]
+#align int.cast_one Int.cast_oneₓ
 
 @[simp, norm_cast]
-theorem cast_neg [AddGroupWithOne R] : ∀ n:ℤ, ((-n : ℤ) : R) = -↑n
+theorem cast_neg : ∀ n, ((-n : ℤ) : R) = -n
   | (0 : ℕ) => by erw [cast_zero, neg_zero]
   | (n + 1 : ℕ) => by erw [cast_ofNat, cast_negSucc]
-  | -[n +1] => by erw [cast_ofNat, cast_negSucc, neg_neg]
+  | -[n+1] => by erw [cast_ofNat, cast_negSucc, neg_neg]
+#align int.cast_neg Int.cast_negₓ
 
-@[simp]
-theorem cast_subNatNat [AddGroupWithOne R] (m n) : ((subNatNat m n : ℤ) : R) = m - n := by
-  unfold Int.subNatNat
+@[simp, norm_cast]
+theorem cast_subNatNat (m n) : ((Int.subNatNat m n : ℤ) : R) = m - n := by
+  unfold subNatNat
   cases e : n - m
-  · simp only [subNatNat, cast_ofNat]
+  · rw [cast_ofNat]
     simp [e, Nat.le_of_sub_eq_zero e]
   · rw [cast_negSucc, Nat.add_one, ← e, Nat.cast_sub <| _root_.le_of_lt <| Nat.lt_of_sub_eq_succ e,
       neg_sub]
-#align int.cast_sub_nat_nat Int.cast_subNatNat
+#align int.cast_sub_nat_nat Int.cast_subNatNatₓ
+
+-- porting note: already exists in `Std`, but still needs `#align`ed
```

Comment:
```suggestion
```
This kind of situation comes up all the time, I don't think it needs a porting note.